assets/js/src/core/modules/document/actions/translations/components/new-translation-modal.tsx

```diff
@@ -0,0 +1,203 @@
+ */
+
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { WindowModal } from '@Pimcore/components/modal/window-modal/window-modal'
+import { ModalFooter } from '@Pimcore/components/modal/footer/modal-footer'
+import { Button } from '@Pimcore/components/button/button'
+import { ManyToOneRelation } from '@Pimcore/components/many-to-one-relation/many-to-one-relation'
+import type { ManyToOneRelationValue } from '@Pimcore/components/many-to-one-relation/many-to-one-relation'
+import { Form } from '@sdk/components'
+import { Input } from '@Pimcore/components/input/input'
+import { Select } from '@Pimcore/components/select/select'
+import { type LanguageOption } from '@Pimcore/modules/document/actions/paste/use-paste'
+import { useSettings } from '@Pimcore/modules/app/settings/hooks/use-settings'
+import { has, isNil, isString } from 'lodash'
+import { useLanguageLookup } from '@Pimcore/modules/translations/hooks/use-language-lookup'
+import { useDocumentGetTranslationParentByLanguageQuery } from '@Pimcore/modules/document/document-api-slice-enhanced'
+import { type Element } from '@Pimcore/modules/element/element-helper'
+
+export interface NewTranslationModalProps {
+  isOpen: boolean
+  useInheritance: boolean
+  onClose: () => void
+  onSubmit: (values: NewTranslationFormValues) => Promise<void>
+  currentDocument: Element | null
+}
+
+export interface NewTranslationFormValues {
+  language: string
+  parent: ManyToOneRelationValue | null
+  title: string
+  navigation: string
+  key: string
+}
+
+export const NewTranslationModal = ({
+  isOpen,
+  useInheritance,
+  onClose,
+  onSubmit,
+  currentDocument
+}: NewTranslationModalProps): React.JSX.Element => {
+  const { t } = useTranslation()
+  const { getDisplayName } = useLanguageLookup()
+  const settings = useSettings()
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [selectedLanguage, setSelectedLanguage] = useState<string>('')
+  const [form] = Form.useForm<NewTranslationFormValues>()
+
+  const languageProperty = (!isNil(currentDocument) && has(currentDocument, 'properties') && Array.isArray(currentDocument?.properties))
+    ? currentDocument.properties?.find(prop => prop.key === 'language')
+    : undefined
+  const currentDocumentLanguage = isString(languageProperty?.data) ? languageProperty.data : ''
+
+  const availableLanguages: LanguageOption[] = (settings.validLanguages ?? [])
+    .filter((locale: string) => locale !== currentDocumentLanguage)
+    .map((locale: string) => ({
+      value: locale,
+      label: `${getDisplayName(locale)} [${locale}]`
+    }))
+
+  const { data: translationParentData } = useDocumentGetTranslationParentByLanguageQuery(
+    {
+      id: currentDocument?.id ?? 0,
+      language: selectedLanguage
+    },
+    { skip: selectedLanguage === '' || isNil(currentDocument?.id) }
+  )
+
+  // Auto-preselect parent when translation parent data is available and contains fullPath
```

Comment:
The auto filling of the parent works on the initial selection of a language, but when we change to another one and then back it doesn't get filled out anymore.